foxtrot-common/src/main/java/com/flipkart/foxtrot/common/query/numeric/BetweenFilter.java

```diff
@@ -80,6 +76,44 @@ public boolean isFilterTemporal() {
         return validationErrors;
     }
 
+    @Override
+    public int hashCode() {
+        int result = getOperator().hashCode();
+        result = 31 * result + getField().hashCode();
+        if(!getField().equals("_timestamp")) {
+            result = result * 21 + (getFrom() == null
+                                    ? 43
+                                    : getFrom().hashCode());
+            result = result * 23 + (getTo() == null
+                                    ? 29
+                                    : getTo().hashCode());
+        }
+        else {
+            result = result * 21 + Long.valueOf(getFrom().longValue() / (long) 30000).hashCode();
```

Comment:
It is not correct to do this for anything other than `last` filter, this behaviour is not correct. We can have an additional flag  which can govern if cached results are acceptable. But for critical cases, people expect correct results and this will definitely not wield one.